ContactTriggerHandler.cls

```diff
@@ -0,0 +1,162 @@
+     * @Created By   : CagriKilic
+     * @Created Date : Jan 06, 2023
+     * @Param con    : List<Contact> - Contact list
+     ******************************************************************/
+    public void processPrimaryAfterInsert(List<Contact> con) {
+        try {
+            List<Id> conIds = new List<Id>();
+            for (Contact c : con) {
+                if (c.Is_Primary_Contact__c == true) {
+                    conIds.add(c.Id);
+                }
+            }
+            if (System.IsBatch() == false && System.isFuture() == false) {
+                processPrimaryAfterInsertAndUpdateAsync(conIds);
+            }
+        } catch (Exception e) {
+            System.debug('Error: ' + e);
+        }
+    }
+    /******************************************************************
+     * @Name         : processPrimaryAfterInsertAndUpdateAsync
+     * @Description  : Async Method of After Insert and Update Trigger
+     * @Created By   : CagriKilic
+     * @Created Date : Jan 06, 2023
+     * @Param con    : List<Id> - Contact Ids
+     ******************************************************************/
+    @future
+    private static void processPrimaryAfterInsertAndUpdateAsync(List<Id> conIds) {
+        try {
+            List<Contact> consToUpdate = new List<Contact>();
+            List<Contact> insertedContacts = [
+                SELECT
+                    Primary_Contact_Phone__c,
+                    Is_Primary_Contact__c,
+                    AccountId,
+                    Phone
+                FROM Contact
+                WHERE Id IN :conIds
+            ];
+
+            List<Contact> oldContact = [
+                SELECT Id, Name, Is_Primary_Contact__c, AccountId
+                FROM Contact
+                WHERE Is_Primary_Contact__c = FALSE
+            ];
+            for (Contact cnt : insertedContacts) {
+                for (Contact c : oldContact) {
+                    if (c.AccountId == cnt.AccountId) {
+                        c.Primary_Contact_Phone__c = cnt.Phone;
+                        consToUpdate.add(c);
+                        if (!consToUpdate.contains(cnt)) {
+                            cnt.Primary_Contact_Phone__c = cnt.Phone;
+                            consToUpdate.add(cnt);
+                        }
+                    }
+                }
+            }
+
+            Database.update(consToUpdate, false);
+        } catch (Exception e) {
+            System.debug('Error: ' + e);
+        }
+    }
+    /******************************************************************
+     * @Name         : validatePrimaryBeforeUpdate
+     * @Description  : Method of Before Update Trigger
+     * @Created By   : CagriKilic
+     * @Created Date : Jan 06, 2023
+     * @Param con    : List<Contact> - Contact list
+     ******************************************************************/
+    public void validatePrimaryBeforeUpdate(List<Contact> con) {
```

Comment:
please add a check in your code if we just update primary from false to true (we do not need to trigger our code if we just updated a Name or smth else, but Primary checkbox was true before update)